docs/developer-docs/backend/rust/10-timers.md

```diff
@@ -0,0 +1,237 @@
+
+```sh
+dfx new --type=rust my_timers --no-frontend
+```
+
+Then, navigate into your project directory by running the command:
+
+```sh
+cd my_timers
+```
+
+:::info
+Note: the following steps assume the terminal is still open and the current directory is `my_timers`.
+:::
+
+## Writing the Cargo.toml file
+
+First, open the `src/my_timers_backend/Cargo.toml` file in a code editor. Replace the existing contents with the following:
+
+```
+The `src/my_timers_backend/Cargo.toml` file:
+
+```toml
+[package]
+name = "my_timers_backend"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+candid = "0.8.2"
+ic-cdk = "0.7.0"
+ic-cdk-macros = "0.6.0"
+ic-cdk-timers = "0.1"
+serde = { version = "1.0", features = ["derive"] }
+```
+
+Save the file. 
+
+## Update the Cargo.toml dependencies
+
+Since we made changes to the Cargo.toml file, run the following command to update the project's dependencies:
+
+```
+cargo update
+```
+
+## Declaring the canister interface
+
+Candid is an interface description language (IDL) for interacting with canisters running on the Internet Computer. Candid files provide a language-independent description of canister interfaces.
+
+To see details about the Candid interface description language syntax, see the [Candid guide](../candid/) or the [Candid crate documentation](https://docs.rs/candid/).
+
+Start by opening the `src/my_timers_backend/my_timers_backend.did` file in a code editor and replace its content with the following:
+
+```candid
+service : (nat64) -> {
+      "counter": () -> (nat64) query;
+}
+```
+
+This code establishes the following:
+
+- `service : (nat64) -> {...}` declares a new service which accepts a single integer argument.
+- `"counter": () -> (nat64) query` declares a canister query entry point named `counter`. The `counter` query takes no arguments `()` and returns an integer `(nat64)`.
+
+Save the file. 
+
+## Implementing the `counter` query
+
+In the previous step the `counter` query is declared as `"counter": () -> (nat64) query`. This step implements it.
+
+In the code editor, open the `src/my_timers_backend/src/lib.rs` file and replace its content with the following:
+
+```rust
+use std::sync::atomic::{AtomicU64, Ordering};
+
+static COUNTER: AtomicU64 = AtomicU64::new(0);
+
+#[ic_cdk::query]
+fn counter() -> u64 {
+    COUNTER.load(Ordering::Relaxed)
+}
+```
+
+This code establishes the following:
+
+- `static COUNTER: AtomicU64 = ...` defines a new global variable called `COUNTER`.
+- `#[ic_cdk::query]` marks the following `counter` function as a `query` entry point, so the function will be exported as `canister_query counter`.
+- `fn counter() -> u64 {...}` defines the query. Just like in the `.did` definition, it takes no arguments and returns `u64`.
+- `COUNTER.load(...)` loads and returns the global `COUNTER` value.
```

Comment:
A note here might be added that `Ordering` is unimportant, for users that don't understand it, and that this is just shorter than but equivalent to `thread_local!` wrapping a `Cell<u64>`.